lib/rex/proto/kerberos/model/krb_cred_info.rb

```diff
@@ -132,6 +159,164 @@ def encode_srealm
+          # @raise [Rex::Proto::Kerberos::Model::Error::KerberosDecodingError] if decoding doesn't succeed
+          # key             [0] EncryptionKey,
+          # prealm          [1] Realm OPTIONAL,
+          # pname           [2] PrincipalName OPTIONAL,
+          # flags           [3] TicketFlags OPTIONAL,
+          # authtime        [4] KerberosTime OPTIONAL,
+          # starttime       [5] KerberosTime OPTIONAL,
+          # endtime         [6] KerberosTime OPTIONAL,
+          # renew-till      [7] KerberosTime OPTIONAL,
+          # srealm          [8] Realm OPTIONAL,
+          # sname           [9] PrincipalName OPTIONAL,
+          # caddr           [10] HostAddresses OPTIONAL
+          def decode_asn1(input)
+            input.value.each do |val|
+              case val.tag
+              when 0
+                self.key = decode_key(val)
+              when 1
+                self.prealm = decode_prealm(val)
+              when 2
+                self.pname = decode_pname(val)
+              when 3
+                self.flags = decode_flags(val)
+              when 4
+                self.auth_time = decode_auth_time(val)
+              when 5
+                self.start_time = decode_start_time(val)
+              when 6
+                self.end_time = decode_end_time(val)
+              when 7
+                self.renew_till = decode_renew_till(val)
+              when 8
+                self.srealm = decode_srealm(val)
+              when 9
+                self.sname = decode_sname(val)
+              when 10
+                ilog "Ignoring optional value #{val.tag} while decoding KrbCredInfo"
+              else
+                raise ::Rex::Proto::Kerberos::Model::Error::KerberosDecodingError, 'Failed to decode KrbCredInfo SEQUENCE'
+              end
+            end
+          end
+
+          # Decodes the key from an OpenSSL::ASN1::ASN1Data
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [EncryptionKey]
+          def decode_key(input)
+            Rex::Proto::Kerberos::Model::EncryptionKey.decode(input.value[0])
+          end
+
+          # Decodes the flags field
+          #
+          # @param input [OpenSSL::ASN1::ASN1Data] the input to decode from
+          # @return [Rex::Proto::Kerberos::Model::KdcOptionFlags]
+          def decode_flags(input)
+            flags = input.value[0].value.unpack1('N')
+            # == OpenSSL::ASN1::BitString
+            #
+            # === Additional attributes
+            # _unused_bits_: if the underlying BIT STRING's
+            # length is a multiple of 8 then _unused_bits_ is 0. Otherwise
+            # _unused_bits_ indicates the number of bits that are to be ignored in
+            # the final octet of the BitString's _value_.
+            unused_bits = input.value[0].unused_bits
+            flags >>= unused_bits
+            Rex::Proto::Kerberos::Model::KdcOptionFlags.new(flags)
```

Comment:
Will we need to apply this change to `enc_kdc_response` as well?

Reply:
I'd like to imagine a response from the KDC would follow the spec and not need this...but I'll add it in anyway